### PR TITLE
UI test for public share with password

### PR DESCRIPTION
**Pull request by paurakhsharma**
## Description
UI test for public share with password

## Related Issue
https://github.com/owncloud/password_policy/issues/126

## How Has This Been Tested?
Locally

## Types of changes
- [x] Tests

## Checklist:
- [x] Acceptance tests added

### CODE REVIEW

File: tests/acceptance/features/bootstrap/WebUISharingContext.php [hunk 1]
```diff
@@ -482,6 +484,50 @@ public function userReactsToShareOfferedByUsingWebUI(
...
+		$this->publicLinkFilesPage->waitForAjaxCallsToStartAndFinish($this->getSession());
+	}
+
+	/**
+	 * @Then the public should not get access to public share file
+	 * 
+	 * @return void
+	 */
+	public function thePublicShouldNotGetAccessToPublicShareFile() {
+		$waringMessage = $this->publicLinkFilesPage->getWarningMessage();
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUISharingContext.php]: s/waringMessage/warningMessage

File: tests/acceptance/features/bootstrap/WebUISharingContext.php [hunk 2]
```diff
@@ -482,6 +484,50 @@ public function userReactsToShareOfferedByUsingWebUI(
...
+	public function thePublicAccessesTheLastCreatedPublicLinkWithWrongPasswordUsingTheWebui($wrongPassword) {
+		$createdPublicLinks = $this->createdPublicLinks;
+		$baseUrl = $this->featureContext->getBaseUrl();
+		$this->publicLinkFilesPage->openPublicShareAuthenticateUrl($createdPublicLinks, $baseUrl);
+		$this->publicLinkFilesPage->enterPublicLinkPassword($wrongPassword);
+		$this->publicLinkFilesPage->waitForAjaxCallsToStartAndFinish($this->getSession());
+	}
+
+	/**
+	 * @Then the public should not get access to public share file
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUISharingContext.php]: Then the public should not get access to the publicly shared file

**Review by phil-davis — COMMENTED** [review]
see minor comments

File: tests/acceptance/features/lib/PublicLinkFilesPage.php [hunk 3]
```diff
@@ -149,6 +152,78 @@ public function moveFileTo(
...
 
+	/**
+	 * enter public link password
+	 *
+	 * @param string $password
+	 *
+	 * @return void
+	 */
+	public function enterPublicLinkPassword($password) {
+		$passwordInputField = $this->findById($this->passwordFieldId);
```
Comment (individual-it) [inline tests/acceptance/features/lib/PublicLinkFilesPage.php]: please check here if the return of `findById()` is `null`

File: tests/acceptance/features/lib/PublicLinkFilesPage.php [hunk 4]
```diff
@@ -149,6 +152,78 @@ public function moveFileTo(
...
+	/**
+	 * waits till at least one Ajax call is active and
+	 * then waits till all outstanding ajax calls finish
+	 *
+	 * @param Session $session
+	 * @param int $timeout_msec
+	 *
+	 * @return void
+	 */
+	public function waitForAjaxCallsToStartAndFinish(
```
Comment (individual-it) [inline tests/acceptance/features/lib/PublicLinkFilesPage.php]: why to reimplement this function?

**Review by individual-it — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/lib/PublicLinkFilesPage.php (hunk 3 above)
Comment (paurakhsharma) [reply tests/acceptance/features/lib/PublicLinkFilesPage.php]: yes, I should do that.

File: tests/acceptance/features/lib/PublicLinkFilesPage.php (hunk 4 above)
Comment (paurakhsharma) [reply tests/acceptance/features/lib/PublicLinkFilesPage.php]: My bad, I should access it through `featureContext`.

**Review by individual-it — APPROVED** [review]

**Review by phil-davis — APPROVED** [review]

**Comment by paurakhsharma** [comment]
Backport on #32741

**Pull request merged**